mlir/test/Dialect/MemRef/invalid.mlir

```diff
@@ -217,6 +217,15 @@ func.func @memref_reinterpret_cast_no_map_but_offset(%in: memref<?xf32>) {
 
 // -----
 
+func.func @memref_reinterpret_cast_offset_mismatch_dynamic(%in: memref<?xf32>, %offset : index) {
+  // expected-error @+1 {{expected result type with offset = -9223372036854775808 instead of 0}}
```

Comment:
THis error message is hard to digest, and is a leak of the sentinal value used to represent dynamic shape. We need a better error message here.

Reply:
done